src/Plugin/facets/facet_source/ListFacetSource.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_list_pages\Plugin\facets\facet_source;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Path\CurrentPathStack;
+use Drupal\facets\FacetSource\SearchApiFacetSourceInterface;
+use Drupal\facets\Plugin\facets\facet_source\SearchApiBaseFacetSource;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Represents a facet source to be used for lists.
+ *
+ * @FacetsFacetSource(
+ *   id = "list_facet_source",
+ *   display_id = "oe_list_pages",
```

Comment:
The display Id references a search API DisplayInterface plugin which we don't have and we will need.

Also, in your deriver, each derivative gets the same display ID as the derivative ID, so you will need a deriver also for the display plugins.